File: view-mode-sample/src/main/java/com/jaynewstrom/viewmodesample/ContentView.java
package com.jaynewstrom.viewmodesample;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.LinearLayout;

import com.jaynewstrom.viewmode.ViewModeView;

import butterknife.ButterKnife;
import butterknife.OnClick;

public final class ContentView extends LinearLayout {

    private ViewModeView viewModeView;

    public ContentView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override protected void onFinishInflate() {
        super.onFinishInflate();
        ButterKnife.bind(this);
    }

    @OnClick(R.id.button) void onButtonClicked() {
        viewModeView.showViewMode(ViewModes.FAILED_TO_LOAD);
    }

    public void setViewModeView(ViewModeView viewModeView) {
        this.viewModeView = viewModeView;
    }
}

File: view-mode-sample/src/main/java/com/jaynewstrom/viewmodesample/ContentView.kt
package com.jaynewstrom.viewmodesample

import android.content.Context
import android.util.AttributeSet
import android.widget.LinearLayout
import butterknife.ButterKnife
import butterknife.OnClick
import com.jaynewstrom.viewmode.ViewModeView

class ContentView(context: Context, attrs: AttributeSet) : LinearLayout(context, attrs) {
    private lateinit var viewModeView: ViewModeView

    override fun onFinishInflate() {
        super.onFinishInflate()
        ButterKnife.bind(this)
    }

    @OnClick(R.id.button) internal fun onButtonClicked() {
        viewModeView.showViewMode(ViewModes.FAILED_TO_LOAD)
    }

    fun setViewModeView(viewModeView: ViewModeView) {
        this.viewModeView = viewModeView
    }
}
